core/events/domain/event.py

```diff
@@ -173,6 +160,10 @@ def to_dict(self):
 class CreateSubmission(Event):
     """Creation of a new :class:`events.domain.submission.Submission`."""
 
+    def validate(self, *args, **kwargs) -> None:
```

Comment:
The rest of the changes in this file are just adding ``validate`` methods for those that didn't have them already.

Reply:
They should raise events.exceptions.InvalidEvent and return None. Before I had Event.valid calling it's subclasses .validate, swallowing the InvalidEvent, and returning bool. The indirection made things difficult to debug